pandas/core/nanops.py

```diff
@@ -453,7 +457,11 @@ def nanall(values, axis=None, skipna: bool = True, mask=None):
     >>> nanops.nanall(s)
     False
     """
-    values, _, _, _, _ = _get_values(values, skipna, fill_value=True, mask=mask)
+    values, _, dtype, _, _ = _get_values(values, skipna, fill_value=True, mask=mask)
+
+    if is_object_dtype(dtype):
+        return all(values)
```

Comment:
can you use `values.all()` here?  that is usually more performant

Reply:
(might also be relevant to the CI failures)

Reply:
> can you use `values.all()` here? that is usually more performant

That's exactly the bug!

Reply:
I see.  what about `np.all(values)` and `np.any(values)`?

Reply:
> I see. what about `np.all(values)` and `np.any(values)`?

Seem to work, I will mark this as resolve if all the test cases passes.

Reply:
@jbrockmendel It doesn't work